src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -722,8 +722,42 @@ export class SearchView extends ViewPane {
 		return this.refreshTree(event);
 	}
 
+	private reselectAfterRefreshTree(beforePreviousSelection: RenderableMatch | null): void {
+		// Selection exists -> no need to reselect as what caused the tree to refresh didn't change the selected match
+		if (this.tree.selectionSize > 0 || !beforePreviousSelection) {
+			return;
+		}
+
+		// Go to the first element of the tree
+		const iterator = this.tree.navigate();
+		let current = iterator.first();
+		// Look for the element before the previous selection in the tree
+		while (!!current) {
+			if (current.id() === beforePreviousSelection.id()) {
+				this.tree.setSelection([current]);
+				this.tree.setFocus([current]);
+				this.tree.reveal(current);
+			}
+			current = iterator.next();
+		}
+	}
+
 	refreshTree(event?: IChangeEvent): void {
 		const collapseResults = this.searchConfig.collapseResults;
+
+		// Get the selection before the refresh
+		const [previous] = this.tree.getSelection();
+		const navigator = this.tree.navigate(previous);
+		let beforePrevious = navigator.previous();
+		// Expand until beforePrevious is a Match
+		while (beforePrevious && !(beforePrevious instanceof Match)) {
+			if (this.tree.isCollapsed(beforePrevious)) {
+				this.tree.expand(beforePrevious);
+			}
+
+			beforePrevious = navigator.previous();
```

Comment:
Before before this step, is the navigator's current selection still on the node that is either now non-existing or isn't a match? Just want to know without running the debugger 😅

Reply:
The navigator.current() result after navigating to selection (line 750) will point to the node that is now non-existing. However, because the tree results aren't updated yet, I can get the previous node (which still exists) and then use it to set the selection when the current is deleted.

Reply:
Makes sense. Can you check the `event` variable for `removed: true` for this so that it doesn't run every time we refresh the tree?